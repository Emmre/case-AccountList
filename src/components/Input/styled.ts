import styled from "styled-components";

export const LabelStyled = styled.label`
  font-family: Nunito Sans;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  letter-spacing: 0.2px;
  color: #77869e;
`;

export const InputStyled = styled.input`
  height: 33px;
  width: 100%;
  border: none;
  border-bottom: 1px solid #bbc3cf;
  margin-bottom: 50px;
  padding-bottom: 8px;
  outline: none;
`;
