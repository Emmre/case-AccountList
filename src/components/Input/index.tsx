import React, { FC, Fragment, InputHTMLAttributes } from "react";
import { InputStyled, LabelStyled } from "./styled";

interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {
  type: string;
  placeholder: string;
  value: string;
  label: string;
}

const Input: FC<IInputProps> = ({
  name,
  label,
  type,
  placeholder,
  value,
  ...rest
}) => (
  <>
    {value && <LabelStyled htmlFor={name}>{label}</LabelStyled>}
    <InputStyled
      id={name}
      type={type}
      placeholder={placeholder}
      value={value}
      {...rest}
    />
  </>
);

export default Input;
