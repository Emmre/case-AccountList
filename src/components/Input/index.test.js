import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Input from "./index";

describe("Input component", () => {
  it("Should be rendering", () => {
    const { container } = render(<Input />);
    const input = container.querySelector("Input");
    expect(input).toBeInTheDocument();
  });

  it("Should match the snapshot", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
