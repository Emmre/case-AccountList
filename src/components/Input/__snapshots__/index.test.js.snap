// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should match the snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-gsDKAQ cKAcLQ"
    value=""
  />
</DocumentFragment>
`;
